### PR TITLE
Fix/get ops between

### LINKED ISSUE #76: Handle all deposit events between parent and current
I think there's a consensus change here that we should make that will also improve the flakiness in the tests by improving the usability of deposits. Rather than looking up only the `ops` at the burn_tip, I think this should look up all the ops between the last mined block and this block.

For example:

```
L1 Chain:

A -> B -> C -> D
1 ------------>2

L2 Chain
```

So L1 block A elects L2 block 1 and L1 block D elects L2 block 2, then the ops that should be included in block 2 would be all the ops in L1 block A, B, and C.

This will make it so that users who submit a deposit will be included in the next L2 block regardless of when that block actually gets elected. This also seems to be the reason some of the tests are flaking.

So, essentially, this is something like:

```rust
let parent_block_burn_block = SortitionDB::get_block_snapshot_consensus(conn, &parent_consensus_hash)?.unwrap().burn_header_hash;
// fetch ops in between `parent_block_burn_block` and `burn_tip` (inclusive).
let deposit_ft_ops = SortitionDB::get_deposit_ft_ops_between(conn, &parent_block_burn_block, &burn_tip)?;
```

Implementing `get_x_between` will probably require walking the sortition db from `burn_tip` backwards, using `parent_sortition_id`.

### PR DESCRIPTION
**Pull request by pavitthrap**
### Description
This PR ensures we get all operations between the current burn tip and the burn hash of the parent hyperchains block. Before, we were only getting operations submitted at the current burn tip. 

### Applicable issues
- fixes #76 

### Checklist
Integration tests now passing reliably locally.

### CODE REVIEW

File: src/chainstate/burn/db/sortdb.rs [hunk 1]
```diff
@@ -2461,6 +2461,31 @@ impl SortitionDB {
...
+            curr_block_id = curr_snapshot.parent_burn_header_hash;
+            let curr_deposit_stx_ops = SortitionDB::get_deposit_stx_ops(conn, &curr_block_id)?;
+            deposit_stx_ops.extend(curr_deposit_stx_ops.into_iter());
+
+            curr_sortition_id = curr_snapshot.parent_sortition_id;
+        }
+
+
+        Ok(deposit_stx_ops)
+    }
```
Comment (kantai) [inline src/chainstate/burn/db/sortdb.rs:2487]: Is it possible to de-dupe the `get_x_ops_between` methods? Maybe something like:
```rust
pub fn get_ops_between<F, Op>(
    conn: &Connection,
    parent_l1_block_id: &BurnchainHeaderHash,
    l1_block_id: &BurnchainHeaderHash,
    get_ops_in_block: F,
    ) -> Result<Vec<Op>, db_error>
where F: Fn(&Connection, &BurnchainHeaderHash) -> Vec<Op>

Comment (kantai) [reply src/chainstate/burn/db/sortdb.rs:2487]: Also, I think the `chain_tip_consensus_hash` parameter is unused.

File: src/chainstate/burn/db/sortdb.rs [hunk 2]
```diff
@@ -2461,6 +2461,31 @@ impl SortitionDB {
         }
     }
 
+    pub fn get_deposit_stx_ops_between(
```
Comment (gregorycoppola) [inline src/chainstate/burn/db/sortdb.rs:2464]: is `parent_l1_block_id` the parent of `l1_block_id`?

Why do we need a loop in that case? Can you make a test that shows when we would need this?

Comment (pavitthrap) [reply src/chainstate/burn/db/sortdb.rs]: `parent_l1_block_id` is the id of the l1 block that the parent hyperchains block is associated with.

Comment (pavitthrap) [reply src/chainstate/burn/db/sortdb.rs]: also, I added a test

**Review by kantai — APPROVED** [review]
LGTM!

**Pull request merged**